# Ensure major(), minor() and makedev() are defined
# Fix for SRM 1.3 having glibc 2.32
# 
# New versions of glibc (since 2.28) and musl (since 1.1.23) actually no longer provide a definition 
# of major by including sys/types.h.
# It's a bit unfortunate since on Linux, we now need to include sys/sysmacros.h
# 
# Solution based on https://github.com/kristapsdz/openrsync/issues/13#issuecomment-559931285
# 
--- src/libudev/libudev.h.orig	2013-05-09 13:45:50.000000000 +0000
+++ src/libudev/libudev.h	2025-09-21 18:44:53.459980291 +0000
@@ -22,6 +22,10 @@
 
 #include <stdarg.h>
 #include <sys/types.h>
+#ifndef major
+// for the case that sys/sysmacros.h is not included within sys/types.h
+#include <sys/sysmacros.h>
+#endif
 #include <sys/stat.h>
 
 #ifdef __cplusplus
--- src/shared/util.h.orig	2013-05-09 13:45:50.000000000 +0000
+++ src/shared/util.h	2025-09-21 18:53:26.206312863 +0000
@@ -33,6 +33,10 @@
 #include <sched.h>
 #include <limits.h>
 #include <sys/types.h>
+#ifndef major
+// for the case that sys/sysmacros.h is not included within sys/types.h
+#include <sys/sysmacros.h>
+#endif
 #include <sys/stat.h>
 #include <dirent.h>
 #include <sys/resource.h>
